schematic/models/validate_attribute.py

```diff
@@ -1027,6 +1240,47 @@ def _gather_set_warnings_errors(
 
         return errors, warnings
 
+    def _remove_non_entry_from_invalid_entry_list(
+        self,
+        invalid_entry: Optional[list[str]],
+        row_num: Optional[list[str]],
+        attribute_name: str,
+    ):
+        """Helper to remove NAs from a list of invalid entries (if applicable, and allowed), remove the row
+        too from row_num. This will make sure errors are not rasied for NA entries unless the value is required.
+        Args:
+            invalid_entry, list[str]: default=None, list of entries in the source manifest where
+                invalid values were located.
+            row_num, list[str[: default=None, list of rows in the source manifest where invalid values were located
+            attribute_name, str: source attribute name
+        Returns:
+            invalid_entry and row_num returned with any NA and corresponding row index value removed, if applicable.
+        """
+        idx_to_remove = []
+        # Check if the current attribute column is required, via the data model
+        if invalid_entry:
+            # Check each invalid entry and determine if it has a value and/or is required.
+            # If there is no entry and its not required, remove the NA value so an error is not raised.
+            for idx, entry in enumerate(invalid_entry):
+                entry_has_value = self.get_entry_has_value(entry, attribute_name)
+                # If there is no value, and is not required, recored the index
+                if not entry_has_value:
+                    idx_to_remove.append(idx)
+
+            # If indices are recorded for NA values to remove, remove them and the corresponding row
+            if idx_to_remove:
+                for idx in sorted(idx_to_remove, reverse=True):
+                    del invalid_entry[idx]
+                    del row_num[idx]
+                # Perform check to make sure length of invalid_entry and row_num is the same. If not that would suggest
+                # there was an issue recording or removing values.
+                if len(invalid_entry) != len(row_num):
```

Comment:
It says `row_num` is optional but it appears the method is assumign its a list

Reply:
I added a check for row_num as I am checking for invalid_entry